FYI, parsing does not seem to work with Marian 1.12 (the `en-fi` project is empty on W&B):
```
[WARNING] Parsing logs from a newer version of Marian (1.12 > 1.10)            
[ERROR] An exception occured parsing /data/mozilla/releng-translations-dev/models/en-fi/opusmt/student-finetuned/train.log: 'perplexity'                       
[ERROR] An exception occured parsing /data/mozilla/releng-translations-dev/models/en-fi/opusmt/student-finetuned/valid.log: 'perplexity'
```

--- a/translations_parser/cli/experiments.py
+++ b/translations_parser/cli/experiments.py
@@ -89,8 +89,6 @@ def parse_experiment(logs_file, project, group, name, metrics_dir=None):
                 config={"logs_file": logs_file},
             )
         ],
-        # There is no Marian context in valid.log files
-        skip_marian_context=name.endswith("_valid"),
     )
     parser.run()
 
@@ -110,15 +108,11 @@ def publish_group_logs(project, group, logs_dir):
 def main():
     args = get_args()
     directory = args.directory
-    file_groups = {path: list(files) for path, files in groupby(directory.glob("**/*.log"), lambda path: path.parent)}
-    total_count = len(file_groups)
-    # Exclude groups missing valid or train log files
+    # Ignore files with a different name than "train.log"
     file_groups = {
-        path: files
-        for path, files in file_groups.items()
-        if ["train.log", "valid.log"] == sorted(f.name for f in files)
+        path: list(files) for path, files in groupby(directory.glob("**/train.log"), lambda path: path.parent)
     }
-    logger.info(f"Reading {len(file_groups)} training data (over {total_count} folders)")
+    logger.info(f"Reading {len(file_groups)} train.log data")
     prefix = os.path.commonprefix([path.parts for path in file_groups])
 
     last_index = None
@@ -148,10 +142,6 @@ def main():
 
         # Publish a run for each file inside that group
         for index, file in enumerate(files, start=1):
-            if file.name == "train.log":
-                name += "_train"
-            elif file.name == "valid.log":
-                name += "_valid"
             # Also publish metric files when available
             metrics_dir = Path("/".join([*prefix, project, group, "evaluation", base_name]))
             if not metrics_dir.is_dir():

--- a/translations_parser/cli/experiments.py
+++ b/translations_parser/cli/experiments.py
@@ -52,7 +52,12 @@ class ExperimentsParser(TrainingParser):
                     continue
                 else:
                     metric = MetricEpoch(up=up)
-                    setattr(metric, metrics_file.stem, value)
+                    setattr(
+                        metric,
+                        # Always prefix metrics to avoid names conflict
+                        f"[metric] {metrics_file.stem}",
+                        value,
+                    )
                     self.training.append(metric)
                     up += 1
         return super().run()

--- a/translations_parser/cli/experiments.py
+++ b/translations_parser/cli/experiments.py
@@ -1,6 +1,7 @@
 import argparse
 import logging
 import os
+from dataclasses import dataclass
 from itertools import groupby
 from pathlib import Path
 
@@ -14,6 +15,47 @@ logging.basicConfig(
     format="[%(levelname)s] %(message)s",
 )
 logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MetricEpoch:
+    """
+    A generic metric extracted from .metric files
+    """
+
+    up: int
+
+
+class ExperimentsParser(TrainingParser):
+    def __init__(self, *args, metrics_dir=None, **kwargs):
+        self.metrics_dir = metrics_dir
+        return super().__init__(*args, **kwargs)
+
+    def run(self):
+        """
+        Add experiment metrics to the training data of the parser
+        so they are also reported by the WandB publisher.
+        """
+        if not self.metrics_dir:
+            return super().run()
+
+        # Add Metrics to the published output based on the name of the file
+        for metrics_file in self.metrics_dir.glob("*.metrics"):
+            with metrics_file.open("r") as f:
+                lines = f.readlines()
+            up = 1
+            for line in lines:
+                # Ignore lines that does not contain a float value
+                try:
+                    value = float(line)
+                except ValueError:
+                    continue
+                else:
+                    metric = MetricEpoch(up=up)
+                    setattr(metric, metrics_file.stem, value)
+                    self.training.append(metric)
+                    up += 1
+        return super().run()
 
 
 def get_args():
@@ -31,23 +73,15 @@ def get_args():
 def parse_experiment(logs_file, project, group, name, metrics_dir=None):
     with logs_file.open("r") as f:
         lines = (line.strip() for line in f.readlines())
-    extra_kwargs = {}
-    if metrics_dir:
-        extra_kwargs.update(
-            {
-                "artifacts": metrics_dir,
-                "artifacts_name": "metrics",
-            }
-        )
-    parser = TrainingParser(
+    parser = ExperimentsParser(
         lines,
+        metrics_dir=metrics_dir,
         publishers=[
             WandB(
                 project=project,
                 name=name,
                 group=group,
                 config={"logs_file": logs_file},
-                **extra_kwargs,
             )
         ],
         # There is no Marian context in valid.log files

--- a/translations_parser/parser.py
+++ b/translations_parser/parser.py
@@ -53,8 +53,9 @@ class TrainingParser:
         self.validation = []
         # Dict mapping (epoch, up) to values parsed on multiple lines
         self._validation_entries = defaultdict(dict)
-        # Marian exection data
+        # Option to read logs directly (skip check for Marian context)
         self.skip_marian_context = skip_marian_context
+        # Marian exection data
         self.version = None
         self.version_hash = None
         self.release_date = None

